### PR TITLE
feat: return an object from pwnedPasswordRange

### LINKED ISSUE #218: Refactor pwnedPasswordRange to resolve with object?
Following on from #24, what if the resolved value from `pwnedPasswordRange` was a simple object mapping the suffixes to the count (`interface PwnedPasswordSuffix { [suffix: string]: number; }`)? E.g. the current test:

https://github.com/wKovacs64/hibp/blob/c92bf7399dcd59b10a817b0207d107ad5e0ff675/src/__tests__/pwnedPasswordRange.test.ts#L14-L27

becomes:

```javascript
      return expect(pwnedPasswordRange('5BAA6')).resolves.toEqual({
        '003D68EB55068C33ACE09247EE4C639306B': 3,
        '1E4C9B93F3F0682250B6CF8331B7EE68FD8': 3303003,
        '01330C689E5D64F660D6947A93AD634EF8F': 1,
      });
```

This would substantially simplify the most common usage, as expressed by the implementation of `pwnedPassword`, from:

https://github.com/wKovacs64/hibp/blob/c92bf7399dcd59b10a817b0207d107ad5e0ff675/src/pwnedPassword.ts#L42-L48

to:

```javascript
  return (
    pwnedPasswordRange(prefix, options)
      .then((arr) => (arr[suffix] || 0))
  );
```

which is also slightly more efficient than an `O(n)` scan of the result array.

I'm happy to PR an implementation if this is a direction people are happy with.

### PR DESCRIPTION
**Pull request by textbook**
Return an object mapping the matching suffix to a count representing the number of occurrences, rather than an array of objects each containing a matching suffix and its count. Closes #218.

BREAKING CHANGE: the return type of `pwnedPasswordRange` is now an object, not an array.

### CODE REVIEW

**Comment by textbook** [comment]
`dist/browser/hibp.umd.js` is right on the 5.3kB size limit; it comes in at 5254 on my machine, which passes, but 5628 in GitHub Actions.

File: src/pwnedPasswordRange.ts [hunk 1]
```diff
@@ -1,18 +1,14 @@
 import { fetchFromApi } from './api/pwnedpasswords';
 
 export interface PwnedPasswordSuffix {
```
Comment (wKovacs64) [inline src/pwnedPasswordRange.ts:3]: We should probably rename `PwnedPasswordSuffix` to `PwnedPasswordSuffixes` since it used to represent a single suffix but now it contains all matching suffixes.

File: src/__tests__/pwnedPassword.test.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ describe('pwnedPassword', () => {
     it('resolves to 0', () => {
       server.use(
         rest.get('*', (_, res, ctx) => {
-          return res(ctx.json(null));
+          return res(ctx.text(EXAMPLE_PASSWORD_HASHES));
```
Comment (wKovacs64) [inline src/__tests__/pwnedPassword.test.ts:22]: Was this change made because there should always be matching suffixes returned? Just curious.

Comment (wKovacs64) [inline src/__tests__/pwnedPassword.test.ts:22]: Was this change made because there should always be matching suffixes returned? Just curious.

**Review by wKovacs64 — CHANGES_REQUESTED** [review]
Looking good, thanks! Can you also run `yarn build:docs` and add the resulting `API.md` to the PR, please?

**Comment by wKovacs64** [comment]
Weird, when I look at the failed workflow, it says all the files passed the size check. Don't worry about that, though - I'll deal with it.

File: src/__tests__/pwnedPassword.test.ts (hunk 2 above)
Comment (textbook) [reply src/__tests__/pwnedPassword.test.ts:22]: When I first read it I was confused by `ctx.json(null)`, so I switched to `ctx.text('')`. Having done that, I thought it would be a more thorough test if there _were_ suffixes in the response, but none matched the test case.

Comment (wKovacs64) [reply src/__tests__/pwnedPassword.test.ts:22]: Gotcha. I originally had that because (I think?) I assumed the remote API responded that way when there were no suffixes, but upon re-reading their docs, it sounds like there will always be data. Good change here, regardless. 👍

**Comment by wKovacs64** [comment]
The increase in size in the UMD build is weirding me out. With these changes, it includes a new ~50 line function (prettified). Guessing it's the introduction of `reduce` usage, but not sure. 😕 

[old-umd.js.txt](https://github.com/wKovacs64/hibp/files/6304484/old-umd.js.txt)
[new-umd.js.txt](https://github.com/wKovacs64/hibp/files/6304485/new-umd.js.txt)

But we're talking about 328 bytes and UMD (does anyone even use that anymore?), so... not going to lose any sleep over it. 🙂

**Review by wKovacs64 — APPROVED** [review]

**Comment by wKovacs64** [comment]
Thanks, @textbook! 🙌
